File: yh_connect/src/yh_connect_pub.cpp
#include "ros/ros.h"
#include"std_msgs/Int32.h"

int main(int argc,char** argv)
{
    ros::init(argc,argv,"yh_connect_pub");//노드이름초기화
    ros::NodeHandle nh;//ros시스템과 통신을위한노드핸들선언
    //퍼블리셔선언
    //패키지(std_msg)의메시지파일(string)을 이용한 퍼블리셔 (pub)룰만든다.
    //토픽은(yh_connect_int)이며,퍼블리서큐(queue)사이즈는(10)이다.
    ros::Publisher pub =nh.advertise<std_msgs::Int32>("yh_connect_int",10);

    ros::Rate loop_rate(4);

    std_msgs::Int32 msg;
    //std_msgs 받을 인자가 Int32로 필요한다 위에int32와다르다(통신을하기위함)

    int cnt=0;

    while(ros::ok())
    {
        msg.data++;
        
        pub.publish(msg);
        loop_rate.sleep();
    }
    return 0;
}

File: yh_connect/scripts/yh_connect_pub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Int32

class ConnectPub:
    def __init__(self):
        self.pub=rospy.Publisher("yh_connect_int",Int32)
        #퍼블리셔선언 노드이름과 자료형이필요하다
        self.msg=Int32()
        #connect_pub의 ㅡmsg instance Int32로 필요한다 위에int32와다르다(통신을하기위함)
        self.loop_rate=rospy.Rate(4)
        #루프속도

    def run(self):
        while not rospy.is_shutdown():
            #꺼지지않으면계속돈다
         #while not rospy.is_shutdown()>rospy.is shutdown()를확인하면서 루프를돌리게된다
        #즉사용자가ctrl+c를눌러작동을멈추기전까지는계속루프를돌린다는의미이다.
            self.pub.publish(self.msg)
            self.msg.data +=1
            self.loop_rate.sleep()

if __name__=="__main__":
    rospy.init_node("yh_connect_pub")
    connect_pub=ConnectPub()
    connect_pub.run()

